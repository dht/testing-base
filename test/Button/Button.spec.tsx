import { ButtonDriver } from './Button.driver';
import Chance from 'chance';

const chance = new Chance();

describe('Button', () => {
    let driver: ButtonDriver;

    beforeAll(() => {
        driver = new ButtonDriver();
    });

    it('should render button', () => {
        const label = chance.word();

        const element = driver.given
            .props({
                title: label,
            })
            .when.rendered();

        const containerClassName = element.get.containerClassName();
        const innerText = element.get.label();

        expect(containerClassName).toContain('Button-container');
        expect(innerText).toBe(label);
    });

    it('should click button', () => {
        const callback = jest.fn();

        driver.given
            .props({
                onClick: callback,
            })
            .when.rendered()
            .when.clicked();

        expect(callback).toHaveBeenCalledTimes(1);
    });
});

describe('Button snapshots', () => {
    let driver: ButtonDriver;

    beforeAll(() => {
        driver = new ButtonDriver();
    });

    it('should match snapshot', () => {
        expect(driver.when.snapshot()).toMatchSnapshot();
    });
});
