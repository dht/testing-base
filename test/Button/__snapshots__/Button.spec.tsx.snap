// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshots should match snapshot 1`] = `
<button
  className="sc-bczRLJ uFwyM Button-container"
  data-testid="Button-container"
  onClick={[Function]}
/>
`;
